lib/board.rb

```diff
@@ -0,0 +1,66 @@
+class Board
+  attr_accessor :cells
+
+    def initialize
+      @cells = Array.new(9, " ")
+    end
+
+    def reset!
+      self.cells.clear
+        self.cells = Array.new(9, " ")
+    end
+
+    def display
+      puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+      puts "-----------"
+      puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+      puts "-----------"
+      puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+    end
+
+    def position(location)
+      new_location = location.to_i - 1
+        self.cells[new_location]
+    end
+
+    def full?
+      self.cells.all? { |token| token == "X" || token == "O" }
+    end
+
+    def turn_count
+      counter = 0
+      self.cells.each do |current_move|
+      if current_move == 'X' || current_move == 'O'
+        counter += 1
+        end
+      end
+      counter
+    end
+
+
+    def taken?(location)
+      new_location = location.to_i - 1
+      new = self.cells[new_location]
+        if new == "X" || new == "O"
+          true
+        else
+          false
+        end
+    end
+
+    def valid_move?(location)
+      new_location = location.to_i - 1
```

Comment:
Use 'and' operator (`&&`) to combine multiple logical expressions and utilize their implicit return value. You can also take advantage of the `taken` method you have defined above. 
```
location.to_i.between?(1,9) && !taken(location)
```